lib/validators/post_validator.rb

```diff
@@ -212,7 +212,7 @@ def acting_user_is_trusted?(post, level = 1)
   end
 
   def private_message?(post)
-    post.topic.try(:private_message?)
+    post.topic.try(:private_message?) || options[:private_message]
```

Comment:
What's happening here? The topic hasn't been built yet, the topic doesn't have the private message archetype, or the post will never have a topic? 🤔

Reply:
yeah this is a nightmare ... the post validator is called prior to topic creator being called. it is a pre-flight check for post creation that happens prior to the big transaction that creates everything. so we have an "orphan" post that does not have any information about what type of post it is. So we pass in `private_message` to the validator to clue it in that it is dealing with a PM so it can use the various PM settings for all the validations it has